client/lib/cart/store/index.js

```diff
@@ -118,8 +111,8 @@ CartStore.dispatchToken = Dispatcher.register( function( payload ) {
 			update( cartItems.removePrivacyFromAllDomains( CartStore.get() ) );
 			break;
 
-		case UpgradesActionTypes.CART_ITEM_ADD:
-			update( cartItems.add( cartItem ) );
+		case UpgradesActionTypes.CART_ITEMS_ADD:
+			update( flow( ...action.cartItems.map( cartItem => cartItems.add( cartItem ) ) ) );
```

Comment:
It would be good idea to have common pattern for function name that returns another function